homeassistant/components/touchline/__init__.py

```diff
@@ -1 +1,60 @@
-"""The touchline component."""
+"""Platform for Roth Touchline floor heating controller."""
+from __future__ import annotations
+
+import logging
+import re
+
+from pytouchline import PyTouchline
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [Platform.CLIMATE]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Roth Touchline from a config entry."""
+    host = entry.data[CONF_HOST]
+    # Remove HTTPS and HTTP from URL
+    pattern = "https?://"
+    host = re.sub(pattern, "", host)
+    host = "http://" + host
```

Comment:
I'm confused what this is meant to accomplish. It looks like it's removing `https://` if it exists and _always_ adding `http://`. Is that intended?

Reply:
Well, in configuration.yaml the value imported is http://x.x.x.x
But in the config flow step, I wanted to allow some flexibility on input, so if the user inputs https, I will strip it and add http (the controller does not have a certificate and will only work with http). The same goes, if the user inputs just the IP-address. I'm a bit bound to how the "pytouchline"-library has implemented httplib2 :-/